apps/ledger-live-desktop/src/renderer/families/tron/StepReceiveFunds.jsx

```diff
@@ -0,0 +1,302 @@
+          </span>
+          <LinkWithExternalIcon
+            style={{ display: "inline-flex" }}
+            onClick={() => openURL(urls.recipientAddressInfo)}
+            label={<Trans i18nKey="common.learnMore" />}
+          />
+        </Text>
+      </Box>
+
+      {renderVerifyUnwrapped({ modelId: device.modelId, type })}
+    </>
+  );
+};
+
+const StepReceiveFunds = (props: StepProps) => {
+  const {
+    isAddressVerified,
+    account,
+    parentAccount,
+    device,
+    onChangeAddressVerified,
+    transitionTo,
+    onResetSkip,
+    verifyAddressError,
+    token,
+    onClose,
+    eventType,
+    currencyName,
+  } = props;
+  const history = useHistory();
+  const receiveStakingFlowConfig = useFeature("receiveStakingFlowConfigDesktop");
+  const mainAccount = account ? getMainAccount(account, parentAccount) : null;
+  invariant(account && mainAccount, "No account given");
+
+  const name = token ? token.name : getAccountName(account);
+  const initialDevice = useRef(device);
+  const address = mainAccount.freshAddress;
+  const [modalVisible, setModalVisible] = useState(false);
+
+  const hideQRCodeModal = useCallback(() => setModalVisible(false), [setModalVisible]);
+  const showQRCodeModal = useCallback(() => setModalVisible(true), [setModalVisible]);
+
+  const onVerify = useCallback(() => {
+    // if device has changed since the beginning, we need to re-entry device
+    if (device !== initialDevice.current || !isAddressVerified) {
+      transitionTo("device");
+    }
+    onChangeAddressVerified(null);
+    onResetSkip();
+  }, [device, onChangeAddressVerified, onResetSkip, transitionTo, isAddressVerified]);
+
+  const onFinishReceiveFlow = useCallback(() => {
+    const id = account?.currency?.id;
+    const dismissModal = global.localStorage.getItem(`${LOCAL_STORAGE_KEY_PREFIX}${id}`) === "true";
+    if (
+      !dismissModal &&
+      receiveStakingFlowConfig?.enabled &&
+      receiveStakingFlowConfig?.params[id]?.enabled
+    ) {
+      track("button_clicked", {
+        button: "continue",
+        page: window.location.hash
+          .split("/")
+          .filter(e => e !== "#")
+          .join("/"),
+        currency: currencyName,
+        modal: "receive",
+        account,
+      });
+
+      transitionTo("stakingFlow");
+      history.push({
+        pathname: `/account/${account.id}`,
+      });
+    } else {
+      onClose();
+    }
+  }, [
+    account,
+    currencyName,
+    history,
+    onClose,
+    receiveStakingFlowConfig?.enabled,
+    receiveStakingFlowConfig?.params,
+    transitionTo,
+  ]);
+
+  return (
+    <>
+      <Box px={2}>
+        <TrackPage
+          category={`Receive Flow${eventType ? ` (${eventType})` : ""}`}
+          name="Step 3"
+          currencyName={currencyName}
+        />
+        {verifyAddressError ? (
+          <ErrorDisplay error={verifyAddressError} onRetry={onVerify} />
+        ) : isAddressVerified === true ? (
```

Comment:
The typing of `isAddressVerified` is kinda weird if we have to check `true` with strict equality 🤔 
Same thing for the `isAddressVerified === false` just under.

I know .jsx is a pain in the ass for simple if/elseif/else, but all of those imbricated ternary operators are making me dizzy 😆 

```jsx
{
  verifyAddressError ? (
    A
   ) : isAddressVerified === true ? (
    B
   ) : isAddressVerified === false ? (
    C
   ) : device ? (
    D
  ) : null // should not happen <--- wut
}
```

And B & C are actually containing ternaries as well 🤯 

Again nothing mandatory, but that would probably help to have a simple function with if/elseif/else outside of the templating imo.

Reply:
Weird stuff indeed. But yeah I'm too afraid at this point to change a thing on this file since most of the code doesn't belongs to me.

I can try if I have not other things to fix tho